src/transformers/modelcard.py

```diff
@@ -438,7 +440,9 @@ def create_model_index(self, metric_mapping):
                         }
                     )
 
-            model_index["results"].append(result)
+            # Remove partial results to avoid the model card being rejected.
+            if "task" in result and "dataset" in result and "metrics" in result:
+                model_index["results"].append(result)
```

Comment:
Would it be worth it to log a message when there's an incomplete `result` ?

Reply:
Will add something!